sdk/java/pulumi/src/test/java/com/pulumi/core/internal/UrnTest.java

```diff
@@ -110,6 +110,19 @@ void testCreateUrnString(
         assertThat(urn).isEqualTo(expected);
     }
 
+    public static Stream<Arguments> testTypeParser() {
+        return Stream.of(
+                arguments("jumbo::ACustomResource", "jumbo:ACustomResource") // tolerate an extra ':'
```

Comment:
not sure if this is undocumented URN feature, but without this codegen test JumboTest fails with the schema it has